### PR TITLE
add msg for publishing of interface values with name and stamp

### PR DESCRIPTION
**Pull request by mamueluth**
This PR is needed in order for the resource manager to be able to publish all the values of the command-/stateInterfaces before writing them to the hw. 
Look at this [PR in the ros2_control repo](https://github.com/ros-controls/ros2_control/pull/1244)

### CODE REVIEW

File: control_msgs/msg/DynamicInterfaceValues.msg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SingleInterfaceValue[] state_interface_values
+SingleInterfaceValue[] command_interface_values
```
Comment (saikishor) [inline control_msgs/msg/DynamicInterfaceValues.msg:2]:
```suggestion
std_msgs/Header header

SingleInterfaceValue[] state_interface_values
SingleInterfaceValue[] command_interface_values
```

File: control_msgs/msg/SingleInterfaceValue.msg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+std_msgs/Header header
+
+# Full qualified interface name e.g. "joint1/position"
+string interface_name
+# Value corresponding to the interface 
+float64 value
```
Comment (saikishor) [inline control_msgs/msg/SingleInterfaceValue.msg:6]:
```suggestion
# Full qualified interface name e.g. "joint1/position"
string interface_name
# Value corresponding to the interface 
float64 value
```
I would suggest moving this header to the DynamicInterfaceValues message as it would make more sense over there. As it will be a duplication when we have a vector.

**Review by saikishor — CHANGES_REQUESTED** [review]

File: control_msgs/msg/DynamicInterfaceValues.msg (hunk 1 above)
Comment (saikishor) [reply control_msgs/msg/DynamicInterfaceValues.msg:2]:
```suggestion
std_msgs/Header header

InterfaceValue state_interface
InterfaceValue command_interface
```

Having values in the names seems redundant
I think with simple `InterfaceValue` message we could achieve the same

File: control_msgs/msg/SingleInterfaceValue.msg (hunk 2 above)
Comment (saikishor) [reply control_msgs/msg/SingleInterfaceValue.msg:6]: if you like my second suggestion, then we can remove this SingleIntyerfaceValue message

**Review by saikishor — CHANGES_REQUESTED** [review]

File: control_msgs/msg/DynamicInterfaceValues.msg (hunk 1 above)
Comment (mamueluth) [reply control_msgs/msg/DynamicInterfaceValues.msg]: already changed this but forgot to push, sry about that

File: control_msgs/msg/InterfaceValue.msg [hunk 3]
```diff
@@ -1,4 +1,4 @@
 # List of resource interface names
 string[] interface_names
 # Values corresponding to the list of interfaces in `interface_names`, [1.0, 0.0] for example
-float64[] values
+float64[] values
```
Comment (bmagyar) [inline control_msgs/msg/InterfaceValue.msg:4]: Please drop this unrelated change

**Review by bmagyar — APPROVED** [review]

File: control_msgs/msg/InterfaceValue.msg (hunk 3 above)
Comment (bmagyar) [reply control_msgs/msg/InterfaceValue.msg]: I dropped it ;)
